browser/brave_federated_learning/brave_federated_learning_service_factory.cc

```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_federated_learning/brave_federated_learning_service_factory.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/browser/brave_browser_process.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/components/brave_federated_learning/brave_federated_learning_service_impl.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/storage_partition.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+namespace brave {
+
+// static
+BraveFederatedLearningService*
+BraveFederatedLearningServiceFactory::GetForProfile(Profile* profile) {
+  if (!brave::IsRegularProfile(profile)) {
+    return nullptr;
+  }
+
+  return static_cast<BraveFederatedLearningService*>(
+      GetInstance()->GetServiceForBrowserContext(profile, true));
+}
+
+// static
+BraveFederatedLearningServiceFactory*
+BraveFederatedLearningServiceFactory::GetInstance() {
+  return base::Singleton<BraveFederatedLearningServiceFactory>::get();
+}
+
+BraveFederatedLearningServiceFactory::BraveFederatedLearningServiceFactory()
+    : BrowserContextKeyedServiceFactory(
+          "BraveFederatedLearningService",
+          BrowserContextDependencyManager::GetInstance()) {}
+
+BraveFederatedLearningServiceFactory::~BraveFederatedLearningServiceFactory() {}
+
+KeyedService* BraveFederatedLearningServiceFactory::BuildServiceInstanceFor(
```

Comment:
what would be the expected behavior for OTR profiles?

Reply:
testing github

Reply:
would this be the expected behavior when the user goes incognito? Should we be collecting the pings in this case? @moritzhaller what is the current behaviour for P3A?

Reply:
if this is referring to incognito, I reckon it would be the callers (e.g. the ads lib calling a logging function of this FL component) responsibility to respect incognito mode and not log anything into the data store (or perform otherwise sensitive operations). It shouldn't affect running this base service, as this service would only be scheduling training runs, server updates or other admin tasks independent of any user behaviour. Not sure if that makes sense @iefremov ?

Reply:
agree that data logging should not be a concern as it depends entirely on other services, but what about activities like collecting pings? Would that be allowed in OTR scenario? That's why I was asking about P3A